### PR TITLE
Use .match? not =~ when match values aren't necessary

### PR DESCRIPTION
**Pull request by tas50**
Autocorrected from RuboCop Performance which is now smart enough to detect when you use the match and when you don't. Using match? does not create any objects so it's slightly faster and uses less memory.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -84,7 +84,7 @@ eval_gemfile("./Gemfile.local") if File.exist?("./Gemfile.local")
 #
 # We copy (and overwrite) these files every time "bundle <exec|install>" is
 # executed, just in case they have changed.
-if RUBY_PLATFORM =~ /mswin|mingw|windows/
+if RUBY_PLATFORM.match?(/mswin|mingw|windows/)
```
Comment (phiggins) [inline Gemfile:87]: String#match? was added in ruby 2.4, does chef support anything older than that?

Comment (tas50) [reply Gemfile:87]: We're 2.6+ now. I've been waiting a while to use this, but the rubocop autocorrect before destroyed code.

File: lib/chef/application/base.rb [hunk 2]
```diff
@@ -363,7 +363,7 @@ def fetch_recipe_tarball(url, path)
     Chef::Log.trace("Download recipes tarball from #{url} to #{path}")
     if File.exist?(url)
       FileUtils.cp(url, path)
-    elsif url =~ URI.regexp
+    elsif url&.match?(URI.regexp)
```
Comment (phiggins) [inline lib/chef/application/base.rb:366]: This nil check could be avoided by flipping these around.

```suggestion
    elsif URI.regexp.match?(url)
```

File: spec/data/lwrp/providers/embedded_resource_accesses_providers_scope.rb [hunk 3]
```diff
@@ -24,5 +24,5 @@ def action_twiddle_thumbs
 end
 
 def generate_new_name(str, &block)
-  "#{str}, #{block.call}"
+  "#{str}, #{yield}"
```
Comment (phiggins) [inline spec/data/lwrp/providers/embedded_resource_accesses_providers_scope.rb:27]: If you're not using the `block` parameter they can be removed from the method's signature.

Comment (tas50) [reply spec/data/lwrp/providers/embedded_resource_accesses_providers_scope.rb:27]: Let me revertthis. I was trying to stick to just the match? fixes in this PR

File: lib/chef/application/base.rb (hunk 2 above)
Comment (tas50) [reply lib/chef/application/base.rb]: Fixed

**Pull request merged**